trainee_template/ablation.amlg

```diff
@@ -0,0 +1,363 @@
+(null
+
+    ;compute the entropy of the influence weights of the context's influential cases
+    ; context_values: optional list of context values to react to
+    ; case_id: optional case id in the model to react to
+    ; context_features: list of context features to react to
+    ; use_case_weights: optional flag default true. if false, case weights will not be used during the react
+    ; weight_feature: optional, default '.case_weight'. name of feature whose values to use as case weights
+    #ComputeInfluenceWeightEntropy
+    (declare
+        (assoc
+            context_values (null)
+            case_id (null)
+            context_features (list)
+            use_case_weights (true)
+            weight_feature autoAblateWeightFeature
+
+            ;internal only
+            react_kwargs (assoc)
+        )
+
+        (if (not use_case_weights)
+            (assign weight_feature ".none"))
+        
+        (assign (assoc
+            react_kwargs (assoc
+                context_features context_features
+                use_case_weights use_case_weights
+                weight_feature weight_feature
+                details (assoc influential_cases (true) )
+            )
+        ))
+        
+        (if 
+            ;if context_values are not null, we are reacting to a new case
+            (not (= context_values (null)))
+            (accum "react_kwargs" (assoc
+                context_values context_values
+            ))
+            ;elif case_id is not null, we are reacting to an existing case
+            (not (= case_id (null)))
+            (accum "react_kwargs" (assoc
```

Comment:
I'd personally prefer this (accum follows the normal styling.
ex: (accum (assoc react_kwargs (assoc blah blah) ))

Reply:
I don't think that's clearer than what's here so (pending intervention from above) i'll keep it the way i've got it